diff -ur ical-2.2.orig/Makefile.in ical-2.2/Makefile.in
--- ical-2.2.orig/Makefile.in	Wed Jul 12 21:24:46 2000
+++ ical-2.2/Makefile.in	Wed Jul 12 21:26:17 2000
@@ -60,7 +60,7 @@
 # Tiny bundled executable uses shared versions of tcl/tk libraries
  TINY_BLIBS = $(LIBS)
 
-       OPTF = -O
+       OPTF =
 
    CXXFLAGS = $(OPTF)
     CXXINCS = -I. -I$(srcdir) -I$(srcdir)/calendar \
diff -ur ical-2.2.orig/calendar/Makefile.in ical-2.2/calendar/Makefile.in
--- ical-2.2.orig/calendar/Makefile.in	Wed Oct 22 01:52:59 1997
+++ ical-2.2/calendar/Makefile.in	Wed Jul 12 21:26:17 2000
@@ -12,7 +12,7 @@
            AR = ar
          ECHO = echo
 
-   CXXFLAGS = -O
+   CXXFLAGS = $(OPTF)
     CXXINCS = -I.. -I$(srcdir) -I$(srcdir)/../types -I$(srcdir)/../time
 
     CXXOBJS = arrays.o calendar.o calfile.o dateset.o item.o\
diff -ur ical-2.2.orig/time/Makefile.in ical-2.2/time/Makefile.in
--- ical-2.2.orig/time/Makefile.in	Sat Feb 10 04:17:47 1996
+++ ical-2.2/time/Makefile.in	Wed Jul 12 21:26:17 2000
@@ -11,7 +11,7 @@
          ECHO = echo
        RANLIB = @RANLIB@
 
-     CXXFLAGS = -O
+     CXXFLAGS = $(OPTF)
       CXXINCS = -I.. -I$(srcdir)
          OBJS = Date.o Month.o Time.o WeekDay.o Year.o parse.o
         TESTS = testparse
diff -ur ical-2.2.orig/types/Makefile.in ical-2.2/types/Makefile.in
--- ical-2.2.orig/types/Makefile.in	Wed Oct 22 01:53:02 1997
+++ ical-2.2/types/Makefile.in	Wed Jul 12 21:27:32 2000
@@ -12,7 +12,7 @@
          ECHO = echo
        RANLIB = @RANLIB@
 
-     CXXFLAGS = -O $(DEFS)
+     CXXFLAGS = $(OPTF) $(DEFS)
       CXXINCS = -I$(srcdir)
          OBJS = bitvec.o hashfuncs.o intset.o
         TESTS = testarray testbv testomap testoset timemap testhash testmap\
